test/cases/db-remove.bats

```diff
@@ -44,6 +44,21 @@ load ../lib/common
 	done
 }
 
+@test "remove partial split package" {
+	local arches=('i686' 'x86_64')
+	local arch
+
+	releasePackage extra pkg-split-a
+	db-update
+
+	for arch in ${arches[@]}; do
+		db-remove extra "${arch}" pkg-split-a1
+	done
+
+	if bsdtar xfO "$FTP_BASE/extra/os/x86_64/extra.db" pkg-split-a1-1-1/desc >/dev/null; then return 1; fi
+	   bsdtar xfO "$FTP_BASE/extra/os/x86_64/extra.db" pkg-split-a2-1-1/desc >/dev/null
```

Comment:
Why doesn't this use the same style as checkRemovedPackageDB? Acknowledging that that function currently checks if the pkgname appears at all, what about extending it to fulfill this use case as well?

Reply:
checkRemovedPackageDB takes a pkgbase, but this test very specifically needs to deal with a specific pkgname in a split package. I suppose doing the check in the same style:

	if bsdtar xfO "$FTP_BASE/extra/os/x86_64/extra.db" | grep pkg-split-a1; then return 1; fi
	   bsdtar xfO "$FTP_BASE/extra/os/x86_64/extra.db" | grep pkg-split-a2

would work, but I don't really see the point.

Reply:
Well, it should use the -xf option to bsdtar instead of the xf format which is strictly for compatibility.

Also why do we check there in every db (for all arches and also, apparently, in the files database), using DBEXT as relevant...

But right here you hardcode one db and check just it. Also, oneliner if statement and incorrectly indented check for pkg-split-a2, plus not quieting the output of grep.

Reply:
(1) Sure, I'll edit it to use the `-`

(2) You're right, hard-coding `x86_64` was a mistake, I should have done this inside of the loop.

(3) Why do the other tests use `$DBEXT` instead of hard-coding `.db`?  pacman itself hard-codes `.db`; we can count on that symlink working for read-only operations.

(4) I didn't quiet grep because it turned out that having the output was helpful to debugging the thing when the test failed.

Reply:
3) I guess using DBEXT **without** stripping the suffix, would almost make sense, but currently I'm not sure why it does what it does.